python/lsst/ip/diffim/makeKernelCandidates.py

```diff
@@ -0,0 +1,127 @@
+                                   "Can only make candidate list from list of afwTable.SourceRecords, "
+                                   f"measAlg.PsfCandidateF or other type with a getSource() method: {e}")
+            candidateList = [c.getSource() for c in candidateList]
+
+        candidateList = diffimTools.sourceToFootprintList(candidateList,
+                                                          templateExposure, scienceExposure,
+                                                          kernelSize,
+                                                          self.kConfig.detectionConfig,
+                                                          self.log)
+        if len(candidateList) == 0:
+            raise RuntimeError("Cannot find any objects suitable for KernelCandidacy")
+
+        return candidateList
+
+    def makeKernelBasisList(self, targetFwhmPix=None, referenceFwhmPix=None,
+                            basisDegGauss=None, basisSigmaGauss=None, metadata=None):
+        """Wrapper to set log messages for
+        `lsst.ip.diffim.makeKernelBasisList`.
+
+        Parameters
+        ----------
+        targetFwhmPix : `float`, optional
+            Passed on to `lsst.ip.diffim.generateAlardLuptonBasisList`.
+            Not used for delta function basis sets.
+        referenceFwhmPix : `float`, optional
+            Passed on to `lsst.ip.diffim.generateAlardLuptonBasisList`.
+            Not used for delta function basis sets.
+        basisDegGauss : `list` of `int`, optional
+            Passed on to `lsst.ip.diffim.generateAlardLuptonBasisList`.
+            Not used for delta function basis sets.
+        basisSigmaGauss : `list` of `int`, optional
+            Passed on to `lsst.ip.diffim.generateAlardLuptonBasisList`.
+            Not used for delta function basis sets.
+        metadata : `lsst.daf.base.PropertySet`, optional
+            Passed on to `lsst.ip.diffim.generateAlardLuptonBasisList`.
+            Not used for delta function basis sets.
+
+        Returns
+        -------
+        basisList: `list` of `lsst.afw.math.kernel.FixedKernel`
+            List of basis kernels.
+        """
+        basisList = makeKernelBasisList(self.kConfig,
+                                        targetFwhmPix=targetFwhmPix,
+                                        referenceFwhmPix=referenceFwhmPix,
+                                        basisDegGauss=basisDegGauss,
+                                        basisSigmaGauss=basisSigmaGauss,
+                                        metadata=metadata)
+        if targetFwhmPix == referenceFwhmPix:
+            self.log.info("Target and reference psf fwhms are equal, falling back to config values")
+        elif referenceFwhmPix > targetFwhmPix:
+            self.log.info("Reference psf fwhm is the greater, normal convolution mode")
+        else:
+            self.log.info("Target psf fwhm is the greater, deconvolution mode")
```

Comment:
It would be nicer to have these log messages occur before calling `makeKernelBasisList`.